src/providers/twitch/TwitchAccount.cpp

```diff
@@ -301,54 +301,80 @@ void TwitchAccount::loadUserstateEmotes(QStringList emoteSetKeys)
         return;
     }
 
-    getIvr()->getBulkEmoteSets(
-        newEmoteSetKeys.join(","),
-        [this](QJsonArray emoteSetArray) {
-            auto emoteData = this->emotes_.access();
-            for (auto emoteSet : emoteSetArray)
-            {
-                auto newUserEmoteSet = std::make_shared<EmoteSet>();
+    // splitting newEmoteSetKeys to batches of 100, because Ivr API endpoint accepts a maximum of 100 emotesets at once
+    constexpr int batchSize = 100;
 
-                IvrEmoteSet ivrEmoteSet(emoteSet.toObject());
+    std::vector<std::vector<QString>> batches;
 
-                newUserEmoteSet->key = ivrEmoteSet.setId;
+    auto newEmoteSetKeysVector = newEmoteSetKeys.toVector();
+    batches.reserve((newEmoteSetKeysVector.size() + 1) / batchSize);
 
-                auto name = ivrEmoteSet.login;
-                name.detach();
-                name[0] = name[0].toUpper();
+    for (int i = 0; i < newEmoteSetKeysVector.size(); i += batchSize)
+    {
+        auto last = std::min(newEmoteSetKeysVector.size(), i + batchSize);
+        batches.emplace_back(newEmoteSetKeysVector.begin() + i,
+                             newEmoteSetKeysVector.begin() + last);
+    }
 
-                newUserEmoteSet->text = name;
-                newUserEmoteSet->type = QString();
-                newUserEmoteSet->channelName = ivrEmoteSet.login;
+    for (const auto &batch : std::move(batches))
+    {
+        QStringList emoteSetBatch;
+
+        for (const auto &el : batch)
+        {
+            emoteSetBatch.push_back(el);
+        }
```

Comment:
If `batches` was `std::vector<QStringList>` (or something along those lines) from the start you wouldn't have to do this.

Reply:
Thank you!

Reply:
This works fine on Qt 5.15, however it doesn't work so well on Qt 5.12, so leaving this workaround for now.